### ISSUE TITLE
VNC support for docker

### ISSUE THREAD

**Issue by noplay** [GNS3/gns3-gui#947]
If a container has is console type set to VNC we need to port forward the container 5900 to the console port and connect the vnc client to it.

**Comment by noplay** [comment]
I try a few day ago. But we need port forward in ubridge if we want do to it. Because with our network interface docker lost port control.

**Comment by AJNOURI** [comment]
Tested VNC client from Windows to remote Ubuntu server (tcp 32768 on the server mapped to tcp 5900 on the container running inside). 
If set the container to start into /bin/bash, the vnc client will connect to container console, and from there start any installed GUI app.

Maybe we could use only VNC port as it gives access to container console and GUI apps started from there.

`docker create -ti --name pc1 -p 127.0.0.1:5900:32768 ajnouri/dockervpc /bin/bash`

![vnc-console](https://cloud.githubusercontent.com/assets/1716020/12740797/13f0039e-c977-11e5-885b-f8c161c8e8cd.png)

![vnc-gui](https://cloud.githubusercontent.com/assets/1716020/12740801/1bb3836c-c977-11e5-89e8-af79e5277c34.png)

**Comment by noplay** [comment]
Sound good. Now we need to found a way to make a port redirect when we inject an ethernet interface inside the container.

**Comment by noplay** [comment]
After all our discussions about injecting stuff in containers I got a solutions for VNC.

I found a way without the need to play with the network. It's pretty common in the docker community to inject their Xsocket inside the container.

At first install on the host machine x11vnc and xvfb it's very common package no worry.

Start Xvfb as fake X server on display 102:
```
Xvfb -nolisten tcp :102
```
GNS3 will need to affect random display port and to keep track of it.

Next I start x11vnc:
```
x11vnc -forever -nopw -display "WAIT:102" -rfbport 6001 -ncache 10
```

Last step is to start the container and inject the X socket as file inside:
``` 
docker run -ti -v /tmp/.X11-unix/X102:/tmp/.X11-unix/X102 -e DISPLAY=:102 xeyes
```

The big advantage is we don't need to play with the network. We can not create trouble in our topologie :) And a container with a graphic application is damn simple:
```
FROM debian
RUN apt-get update
RUN apt-get install -qqy x11-apps
CMD xeyes
```
